game/plugin/navigation/door/src/door.plugin.kts

```diff
@@ -0,0 +1,212 @@
+import org.apollo.game.action.DistancedAction
+import org.apollo.game.model.entity.Player
+import org.apollo.game.model.entity.obj.DynamicGameObject
+import org.apollo.game.model.entity.EntityType
+import org.apollo.game.model.entity.obj.GameObject
+import org.apollo.game.model.Position
+import org.apollo.game.model.World
+import org.apollo.game.model.event.PlayerEvent
+import org.apollo.game.message.impl.ObjectActionMessage
+import org.apollo.net.message.Message
+
+enum class Orientation(val value: Int) {
+    WEST(0), NORTH(1), EAST(2), SOUTH(3);
+    companion object {
+        fun from(findValue: Int): Orientation = Orientation.values().first { it.value == findValue }
```

Comment:
I think `findValue` would be more accurately named `ordinal`.

Reply:
Please use `Direction` and `Direction.WNES` instead.

Reply:
Can you enlighten me on how I would get direction from gameObject orientation integer?

As currently, it is causing me issues for this refactoring. The following code getting orientation/direction by position from the gameObject is causing issues during removal of an entity. Not seeing how it is related. @garyttierney 
```
Direction.fromDeltas(position.x, position.y)
```

```
private fun movePosition(): Position {
        val position = gameObject.position
        System.out.println("DIRECTION" + Direction.fromDeltas(position.x, position.y).toInteger())
        return when(Direction.fromDeltas(position.x, position.y)) {
            Direction.WEST -> position.step(1, Direction.WEST)
            Direction.EAST -> position.step(1, Direction.EAST)
            Direction.NORTH -> position.step(1, Direction.NORTH)
            Direction.SOUTH -> position.step(1, Direction.SOUTH)
            else -> return position
        }

    }
```

Reply:
```kotlin
Direction.WNES[object.orientation]
```

That could be made clearer.